test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpMetricsExporterTests.cs

```diff
@@ -704,6 +706,50 @@ public void TestHistogramToOtlpMetric(string name, string description, string un
             Assert.Empty(dataPoint.Exemplars);
         }
 
+        [Theory]
+        [InlineData("cumulative", MetricReaderTemporalityPreference.Cumulative)]
+        [InlineData("Cumulative", MetricReaderTemporalityPreference.Cumulative)]
+        [InlineData("CUMULATIVE", MetricReaderTemporalityPreference.Cumulative)]
+        [InlineData("delta", MetricReaderTemporalityPreference.Delta)]
+        [InlineData("Delta", MetricReaderTemporalityPreference.Delta)]
+        [InlineData("DELTA", MetricReaderTemporalityPreference.Delta)]
+        public void TestTemporalityPreferenceConfiguration(string configValue, MetricReaderTemporalityPreference expectedTemporality)
+        {
+            var configData = new Dictionary<string, string> { ["OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE"] = configValue };
```

Comment:
nit: Use the constant `OtlpMetricExporterExtensions.OtlpMetricExporterTemporalityPreferenceEnvVarKey` here (I'm assuming the tests can see internals)

Reply:
This was done on purpose. If someone accidentally changes the environment variable key name in the `src` file, then our tests should complain about it. If I use the `const string` defined in the `src` file then our tests would simply pass even when the value of the key is changed.

Reply:
I don't necessarily disagree with your logic, but it does break with all other envvar tests that I can recall seeing. Ex: https://github.com/open-telemetry/opentelemetry-dotnet/blob/e7f7cd68759d81aa3a34cb88d1b2590e09c21e89/test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsTests.cs#L62-L65

I would probably go with the constant to be consistent and follow DRY principle. If someone does have a valid reason to change it they should only have to do it in one spot.

But I made this a "nit" comment because I figured you might disagree 🤣 Up to you!

Reply:
I think we should think about what do we want to test? If we want to test, that a user a can configure something by providing a value for a particular string in the config, then our tests shouldn't be using the same const as the src file. Our tests would be too dynamic for their own good. 😄